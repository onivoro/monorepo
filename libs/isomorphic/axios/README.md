# @onivoro/isomorphic-axios

Lightweight Axios wrapper for creating configured HTTP clients with built-in error handling and retry logic. This package provides utilities for creating Axios instances with custom interceptors and integrating with OpenAPI-generated client classes.

## Installation

```bash
npm install @onivoro/isomorphic-axios
```

## Features

- **Simple Axios Instance Creation**: Quick setup with error handlers and header injection
- **Advanced Factory Pattern**: Comprehensive configuration with retry logic and status-specific handlers
- **OpenAPI Integration**: Seamless integration with auto-generated API client classes
- **Built-in Retry Logic**: Configurable retry mechanism with delays
- **TypeScript Support**: Full type definitions for all configurations

## API Reference

### Core Functions

#### `createAxiosInstance(config: TApiConfig): AxiosInstance`

Creates a basic Axios instance with request/response interceptors for headers and error handling.

```typescript
import { createAxiosInstance } from '@onivoro/isomorphic-axios';

const client = createAxiosInstance({
  apiUrl: 'https://api.example.com',
  addHeaders: (req) => ({
    'Authorization': `Bearer ${getToken()}`,
    'X-API-Key': 'my-api-key'
  }),
  on400: (response) => {
    console.error('Bad request:', response.data);
  },
  on401: (response) => {
    // Handle unauthorized
    window.location.href = '/login';
  },
  on403: (response) => {
    console.error('Forbidden:', response.data);
  },
  on500: (response) => {
    console.error('Server error:', response.data);
  },
  onRequest: (request) => {
    console.log('Request:', request.url);
  },
  onResponse: (response) => {
    console.log('Response:', response.status);
  },
  onError: (response) => {
    console.error('Error:', response);
  }
});

// Use the client
const data = await client.get('/users');
```

**Note**: Error handlers (on400, on401, etc.) do not throw errors by default. If an error handler is not provided for a status code, the error will be thrown.

#### `createApi<TApi>(DefaultApi: ConstructorFunction<TApi>, config: TApiConfig): TApi`

Creates an instance of an OpenAPI-generated client class with a configured Axios instance.

```typescript
import { createApi } from '@onivoro/isomorphic-axios';
import { UserApi } from './generated/api';

const userApi = createApi(UserApi, {
  apiUrl: 'https://api.example.com',
  addHeaders: (req) => ({
    'Authorization': `Bearer ${getToken()}`
  }),
  on401: (response) => {
    // Handle unauthorized
    redirectToLogin();
  }
});

// Use the generated API methods
const users = await userApi.getUsers();
const user = await userApi.createUser({ name: 'John Doe' });
```

**Note**: This function is designed to work with OpenAPI client generators that accept `(configuration, basePath, axiosInstance)` as constructor parameters.

#### `axiosInstanceFactory<TData>(params): AxiosInstance`

Advanced factory for creating Axios instances with retry logic and granular control over headers and error handling.

```typescript
import { axiosInstanceFactory } from '@onivoro/isomorphic-axios';

const client = axiosInstanceFactory({
  headerSetters: {
    'Authorization': (req) => `Bearer ${getToken()}`,
    'X-Request-ID': (req) => generateRequestId()
  },
  errorHandlers: {
    401: async (err, response) => {
      // Handle unauthorized
      await refreshToken();
      throw err; // Retry will happen if configured
    },
    0: async (err, response) => {
      // Default handler for unhandled status codes
      console.error('Request failed:', err);
      throw err;
    }
  },
  beforeRetryHandlers: {
    429: async (err, response) => {
      // Wait before retrying rate-limited requests
      const retryAfter = response?.headers['retry-after'];
      await sleep(retryAfter * 1000);
    }
  },
  axiosConfigOverride: {
    timeout: 10000,
    retry: 3,
    retryDelay: 1000
  }
});
```

**Important**: 
- Error handlers must either return a value or throw an error
- The `0` key in errorHandlers/beforeRetryHandlers acts as a default handler
- Retry logic is built into the interceptor

#### `retryAxiosCall<TData>(err: any, instance: AxiosInstance): Promise<any>`

Utility function for manually implementing retry logic. Used internally by `axiosInstanceFactory`.

```typescript
import { retryAxiosCall } from '@onivoro/isomorphic-axios';

// Custom retry implementation
instance.interceptors.response.use(
  response => response,
  async (error) => {
    if (error.config && error.config.retry > 0) {
      return await retryAxiosCall(error, instance);
    }
    throw error;
  }
);
```

### Constants

#### `defaultAxiosConfig: IRetryConfig<any>`

Default configuration for retry behavior:

```typescript
{
  retry: 3,
  retryDelay: 1000
}
```

### Types

#### `TApiConfig`

Configuration type for `createAxiosInstance` and `createApi`:

```typescript
type TApiConfig = {
  apiUrl: string;
  uiUrl?: string;
  addHeaders?: (req: any) => Record<string, string>;
  on400?: (response: any) => any;
  on401?: (response: any) => any;
  on403?: (response: any) => any;
  on500?: (response: any) => any;
  onRequest?: (request: any) => any;
  onResponse?: (response: any) => any;
  onError?: (response: any) => any;
};
```

#### `IRetryConfig<TData>`

Extends Axios request config with retry settings:

```typescript
interface IRetryConfig<TData> extends AxiosRequestConfig<TData> {
  retry?: number;
  retryDelay?: number;
}
```

#### `TErrorHandler<TData>`

Type for error handling functions:

```typescript
type TErrorHandler<TData> = (
  err?: any,
  res?: AxiosResponse<TData, IRetryConfig<TData>>
) => Promise<TData>;
```

#### `THeaderSetterMap<TData>`

Map of header names to functions that compute header values:

```typescript
type THeaderSetterMap<TData> = Record<string, (req: IRetryConfig<TData>) => string>;
```

#### `TErrorHandlerMap<TData>`

Map of HTTP status codes to error handlers:

```typescript
type TErrorHandlerMap<TData> = Record<number, TErrorHandler<TData>>;
```

## Common Usage Patterns

### Simple API Client

```typescript
import { createAxiosInstance } from '@onivoro/isomorphic-axios';

const api = createAxiosInstance({
  apiUrl: process.env.API_URL,
  addHeaders: () => ({
    'Authorization': `Bearer ${localStorage.getItem('token')}`
  }),
  on401: () => {
    localStorage.removeItem('token');
    window.location.href = '/login';
  }
});

// Make requests
const users = await api.get('/users');
const user = await api.post('/users', { name: 'Jane Doe' });
```

### API Client with Retry Logic

```typescript
import { axiosInstanceFactory, defaultAxiosConfig } from '@onivoro/isomorphic-axios';

const apiWithRetry = axiosInstanceFactory({
  headerSetters: {
    'Authorization': () => `Bearer ${getToken()}`
  },
  errorHandlers: {
    0: async (err) => {
      console.error('Request failed:', err.message);
      throw err;
    }
  },
  beforeRetryHandlers: {
    0: async (err) => {
      console.log(`Retrying request... ${err.config.retry} attempts left`);
    }
  },
  axiosConfigOverride: {
    ...defaultAxiosConfig,
    baseURL: 'https://api.example.com',
    retry: 5,
    retryDelay: 2000
  }
});
```

### OpenAPI Client Integration

```typescript
import { createApi } from '@onivoro/isomorphic-axios';
import { DefaultApi } from './generated';

const api = createApi(DefaultApi, {
  apiUrl: process.env.API_BASE_URL,
  addHeaders: (req) => {
    const headers: Record<string, string> = {
      'Content-Type': 'application/json'
    };
    
    const token = getAuthToken();
    if (token) {
      headers['Authorization'] = `Bearer ${token}`;
    }
    
    return headers;
  },
  onError: (response) => {
    logError('API Error', response);
  }
});

// Use typed methods from generated client
const result = await api.getUserById(123);
```

### Environment-Specific Configuration

```typescript
import { createAxiosInstance } from '@onivoro/isomorphic-axios';

const isDev = process.env.NODE_ENV === 'development';

const client = createAxiosInstance({
  apiUrl: isDev ? 'http://localhost:3000' : 'https://api.prod.com',
  addHeaders: (req) => ({
    'Authorization': `Bearer ${getToken()}`,
    ...(isDev && { 'X-Debug': 'true' })
  }),
  onRequest: isDev ? (req) => {
    console.log(`[API] ${req.method?.toUpperCase()} ${req.url}`);
  } : undefined,
  onResponse: isDev ? (res) => {
    console.log(`[API] Response ${res.status}`);
  } : undefined,
  onError: (response) => {
    if (isDev) {
      console.error('[API] Error:', response);
    }
    // Log to error tracking service in production
    if (!isDev) {
      errorTracker.log(response);
    }
  }
});
```

## Important Notes

1. **Error Handling**: In `createAxiosInstance`, error handlers (on400, on401, etc.) suppress errors by default. If you need to propagate errors, you must throw them explicitly in your handler.

2. **Retry Configuration**: When using `axiosInstanceFactory`, retry configuration should be passed in `axiosConfigOverride`. The retry counter decrements with each attempt.

3. **Header Functions**: Header setter functions receive the request configuration object, allowing dynamic header generation based on request context.

4. **OpenAPI Integration**: The `createApi` function expects generated API classes that follow the pattern of accepting configuration, basePath, and axios instance as constructor parameters.

## License

This library is licensed under the MIT License. See the LICENSE file in this package for details.