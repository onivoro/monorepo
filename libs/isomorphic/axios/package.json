{
  "name": "@onivoro/isomorphic-axios",
  "version": "24.0.3",
  "type": "commonjs",
  "main": "./src/index.js",
  "types": "./src/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/onivoro/monorepo.git"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "axios": "~1.7.7"
  }
}
