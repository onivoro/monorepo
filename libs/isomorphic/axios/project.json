{
  "name": "lib-isomorphic-axios",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/isomorphic/axios/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/isomorphic/axios",
        "main": "libs/isomorphic/axios/src/index.ts",
        "tsConfig": "libs/isomorphic/axios/tsconfig.lib.json",
        "assets": [
          "libs/isomorphic/axios/README.md",
          "libs/isomorphic/axios/package.json"
        ],
        "declaration": true
      }
    }
  },
  "tags": []
}
