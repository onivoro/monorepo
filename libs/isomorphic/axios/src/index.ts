export * from './lib/constants/default-axios-config.constant';

export * from './lib/functions/axios-instance-factory.function';
export * from './lib/functions/create-api.function';
export * from './lib/functions/create-axios-instance.function';
export * from './lib/functions/retry-axios-call.function';

export * from './lib/types/api-config.type';
export * from './lib/types/error-handler-map.type';
export * from './lib/types/error-handler.type';
export * from './lib/types/header-setting-map.type';
export * from './lib/types/retry-config.interface';
