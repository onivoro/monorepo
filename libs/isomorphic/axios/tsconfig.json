{
  "extends": "../../../tsconfig.isomorphic.json",
  "compilerOptions": {
    "module": "CommonJS",
    "outDir": "../../../dist/out-tsc"
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    }
  ]
}
