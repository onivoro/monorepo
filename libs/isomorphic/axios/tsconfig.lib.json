{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "types": ["node", "vite/client"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["src/**/*.spec.ts", "src/**/*.test.ts"]
}
