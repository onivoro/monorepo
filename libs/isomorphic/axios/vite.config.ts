import { viteConfigFactory } from '@onivoro/onix';

export default viteConfigFactory({
    root: __dirname,
    tsconfigPath: 'tsconfig.lib.json',
    cacheDir: '../../../node_modules/.vite/libs/isomorphic/axios',
    outDir: '../../../dist/libs/isomorphic/axios',
});
