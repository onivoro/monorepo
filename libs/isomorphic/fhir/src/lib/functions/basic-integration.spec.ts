import axios from 'axios';
import { normalizeFhirBundle } from './normalize-fhir-bundle.function';
import { IFhirBundle } from '../interfaces/fhir-resources.interface';

describe('Basic FHIR Integration Test', () => {
  const FHIR_BASE_URL = 'http://hapi.fhir.org/baseR4';
  const timeout = 30000;

  beforeAll(() => {
    axios.defaults.timeout = timeout;
  });

  it('should successfully fetch and normalize patient data from HAPI FHIR server', async () => {
    // Test server availability first
    let serverAvailable = true;
    try {
      const healthCheck = await axios.get(`${FHIR_BASE_URL}/metadata?_format=json`, { timeout: 5000 });
      expect(healthCheck.status).toBe(200);
      expect(healthCheck.data.resourceType).toBe('CapabilityStatement');
    } catch (error) {
      console.warn('FHIR server health check failed:', error);
      serverAvailable = false;
    }

    if (!serverAvailable) {
      console.warn('Skipping integration test - FHIR server not available');
      return;
    }

    // Fetch patient data
    const response = await axios.get(`${FHIR_BASE_URL}/Patient?_count=3&_format=json`);

    expect(response.status).toBe(200);
    expect(response.data).toHaveProperty('resourceType', 'Bundle');

    const bundle: IFhirBundle = response.data;
    expect(bundle.resourceType).toBe('Bundle');

    // Handle case where server returns empty results
    if (bundle.entry && bundle.entry.length > 0) {
      expect(Array.isArray(bundle.entry)).toBe(true);
    }

    // Test normalization function with real data
    const normalizedData = normalizeFhirBundle(bundle);
    expect(normalizedData).toBeDefined();
    expect(normalizedData.bundles.length).toBe(1);
    expect(normalizedData.bundleEntries.length).toBeGreaterThanOrEqual(0);

    // If there are patients in the response, validate they are normalized correctly
    if (bundle.entry && bundle.entry.length > 0) {
      const patientEntries = bundle.entry.filter(entry =>
        entry.resource?.resourceType === 'Patient'
      );

      if (patientEntries.length > 0) {
        expect(normalizedData.patients.length).toBeGreaterThan(0);

        // Validate first patient structure
        const firstPatient = normalizedData.patients[0];
        expect(firstPatient.id).toBeDefined();
        expect(firstPatient.resourceId).toBeDefined();
      }
    }

    console.log(`✅ Successfully processed FHIR bundle with ${bundle.entry?.length || 0} entries`);
    console.log(`✅ Normalized into ${normalizedData.patients.length} patients`);
  }, timeout);

  it('should handle empty or small result sets gracefully', async () => {
    try {
      // Try to find a resource that may not exist
      const response = await axios.get(`${FHIR_BASE_URL}/Patient?name=ThisNameProbablyDoesNotExist&_count=1&_format=json`);

      expect(response.status).toBe(200);
      const bundle: IFhirBundle = response.data;
      expect(bundle.resourceType).toBe('Bundle');

      // Should handle empty results gracefully
      const normalizedData = normalizeFhirBundle(bundle);
      expect(normalizedData.bundles.length).toBe(1);

      if (bundle.entry && bundle.entry.length === 0) {
        console.log('✅ Handled empty result set gracefully');
      } else {
        console.log('✅ Handled small result set gracefully');
      }
    } catch (error: any) {
      // Some servers might return errors for invalid searches
      if (error.response?.status >= 400 && error.response?.status < 500) {
        console.log('✅ Handled server error gracefully');
      } else {
        throw error;
      }
    }
  }, timeout);

  it('should validate library exports are working correctly', () => {
    // Test that the function can be imported and called
    expect(typeof normalizeFhirBundle).toBe('function');

    // Test with a minimal valid bundle
    const minimalBundle: IFhirBundle = {
      resourceType: 'Bundle',
      type: 'searchset',
      entry: []
    };

    const result = normalizeFhirBundle(minimalBundle);
    expect(result).toBeDefined();
    expect(result.bundles.length).toBe(1);
    expect(result.bundleEntries.length).toBe(0);

    console.log('✅ Library exports are working correctly');
  });
});
