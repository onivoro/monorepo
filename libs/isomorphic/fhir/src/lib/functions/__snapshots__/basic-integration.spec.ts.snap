// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic FHIR Integration Test should handle empty or small result sets gracefully 1`] = `
{
  "addresses": [],
  "allergyIntolerances": [],
  "allergyManifestations": [],
  "allergyReactions": [],
  "appointmentParticipants": [],
  "appointments": [],
  "attachments": [],
  "bundleEntries": [],
  "bundles": [
    {
      "createdAt": 2025-08-04T15:45:29.821Z,
      "id": "36160a4b-8f13-4e3a-a23e-ea85113817ef",
      "resourceId": "2fbd871b-bd6c-46c8-b611-901b5d5d104b",
      "timestamp": undefined,
      "total": 0,
      "type": "searchset",
    },
  ],
  "carePlanActivities": [],
  "carePlans": [],
  "careTeamParticipants": [],
  "careTeams": [],
  "codeableConcepts": [],
  "codings": [],
  "conditions": [],
  "diagnosticReportResults": [],
  "diagnosticReports": [],
  "encounterParticipants": [],
  "encounters": [],
  "immunizations": [],
  "locations": [],
  "medicationDosages": [],
  "medicationRequests": [],
  "notes": [],
  "observationComponents": [],
  "observations": [],
  "organizationTypes": [],
  "organizations": [],
  "patientContacts": [],
  "patientIdentifiers": [],
  "patientNames": [],
  "patients": [],
  "practitionerQualifications": [],
  "practitioners": [],
  "procedurePerformers": [],
  "procedures": [],
  "references": [],
  "telecoms": [],
}
`;

exports[`Basic FHIR Integration Test should successfully fetch and normalize patient data from HAPI FHIR server 1`] = `
{
  "addresses": [],
  "allergyIntolerances": [],
  "allergyManifestations": [],
  "allergyReactions": [],
  "appointmentParticipants": [],
  "appointments": [],
  "attachments": [],
  "bundleEntries": [
    {
      "bundleId": "bc8ba102-f417-4af4-8837-8ea29c7b56e4",
      "createdAt": 2025-08-04T15:45:28.837Z,
      "fullUrl": "https://hapi.fhir.org/baseR4/Patient/44821623",
      "id": "3fadb2b0-4f6d-4cf3-80f9-9a8ff02bc3fd",
      "resourceId": "44821623",
      "resourceType": "Patient",
      "searchMode": "match",
      "searchScore": undefined,
    },
    {
      "bundleId": "bc8ba102-f417-4af4-8837-8ea29c7b56e4",
      "createdAt": 2025-08-04T15:45:28.837Z,
      "fullUrl": "https://hapi.fhir.org/baseR4/Patient/10971143",
      "id": "bb64a468-b3f3-4f8f-a756-ac8cd0ffd253",
      "resourceId": "10971143",
      "resourceType": "Patient",
      "searchMode": "match",
      "searchScore": undefined,
    },
    {
      "bundleId": "bc8ba102-f417-4af4-8837-8ea29c7b56e4",
      "createdAt": 2025-08-04T15:45:28.837Z,
      "fullUrl": "https://hapi.fhir.org/baseR4/Patient/10971146",
      "id": "68f996fb-ba73-45bb-bf84-658725354bd7",
      "resourceId": "10971146",
      "resourceType": "Patient",
      "searchMode": "match",
      "searchScore": undefined,
    },
  ],
  "bundles": [
    {
      "createdAt": 2025-08-04T15:45:28.837Z,
      "id": "bc8ba102-f417-4af4-8837-8ea29c7b56e4",
      "resourceId": "f3a91003-0dc8-43b9-a61b-0662261c516f",
      "timestamp": undefined,
      "total": undefined,
      "type": "searchset",
    },
  ],
  "carePlanActivities": [],
  "carePlans": [],
  "careTeamParticipants": [],
  "careTeams": [],
  "codeableConcepts": [],
  "codings": [],
  "conditions": [],
  "diagnosticReportResults": [],
  "diagnosticReports": [],
  "encounterParticipants": [],
  "encounters": [],
  "immunizations": [],
  "locations": [],
  "medicationDosages": [],
  "medicationRequests": [],
  "notes": [],
  "observationComponents": [],
  "observations": [],
  "organizationTypes": [],
  "organizations": [],
  "patientContacts": [],
  "patientIdentifiers": [],
  "patientNames": [
    {
      "createdAt": 2025-08-04T15:45:28.837Z,
      "family": "Doe",
      "given": "John",
      "id": "cf6f75ef-048c-459c-b6c5-aab938fa557e",
      "patientId": "88cc98d1-30c3-4425-a0cd-7d4efa60cd3c",
      "periodEnd": undefined,
      "periodStart": undefined,
      "prefix": undefined,
      "suffix": undefined,
      "text": undefined,
      "use": "official",
    },
    {
      "createdAt": 2025-08-04T15:45:28.837Z,
      "family": undefined,
      "given": undefined,
      "id": "f064a035-c005-485d-8dd8-ecbd7bbda139",
      "patientId": "c94d94cd-074d-4b90-8380-3291ea418124",
      "periodEnd": undefined,
      "periodStart": undefined,
      "prefix": undefined,
      "suffix": undefined,
      "text": "Ash",
      "use": undefined,
    },
    {
      "createdAt": 2025-08-04T15:45:28.837Z,
      "family": undefined,
      "given": undefined,
      "id": "2b8063b4-48f4-456d-a612-4fa801467b7b",
      "patientId": "d0e53f46-7b40-47ce-8917-cef1d9d8db8d",
      "periodEnd": undefined,
      "periodStart": undefined,
      "prefix": undefined,
      "suffix": undefined,
      "text": "Ash",
      "use": undefined,
    },
  ],
  "patients": [
    {
      "active": undefined,
      "birthDate": "1980-01-01",
      "createdAt": 2025-08-04T15:45:28.837Z,
      "deceased": undefined,
      "deceasedDateTime": undefined,
      "gender": "male",
      "id": "88cc98d1-30c3-4425-a0cd-7d4efa60cd3c",
      "managingOrganizationId": undefined,
      "maritalStatus": undefined,
      "multipleBirth": undefined,
      "multipleBirthInteger": undefined,
      "resourceId": "44821623",
      "updatedAt": 2025-08-04T15:45:28.837Z,
    },
    {
      "active": undefined,
      "birthDate": undefined,
      "createdAt": 2025-08-04T15:45:28.837Z,
      "deceased": undefined,
      "deceasedDateTime": undefined,
      "gender": undefined,
      "id": "c94d94cd-074d-4b90-8380-3291ea418124",
      "managingOrganizationId": undefined,
      "maritalStatus": undefined,
      "multipleBirth": undefined,
      "multipleBirthInteger": undefined,
      "resourceId": "10971143",
      "updatedAt": 2025-08-04T15:45:28.837Z,
    },
    {
      "active": undefined,
      "birthDate": undefined,
      "createdAt": 2025-08-04T15:45:28.837Z,
      "deceased": undefined,
      "deceasedDateTime": undefined,
      "gender": undefined,
      "id": "d0e53f46-7b40-47ce-8917-cef1d9d8db8d",
      "managingOrganizationId": undefined,
      "maritalStatus": undefined,
      "multipleBirth": undefined,
      "multipleBirthInteger": undefined,
      "resourceId": "10971146",
      "updatedAt": 2025-08-04T15:45:28.837Z,
    },
  ],
  "practitionerQualifications": [],
  "practitioners": [],
  "procedurePerformers": [],
  "procedures": [],
  "references": [],
  "telecoms": [],
}
`;

exports[`Basic FHIR Integration Test should validate library exports are working correctly 1`] = `
{
  "addresses": [],
  "allergyIntolerances": [],
  "allergyManifestations": [],
  "allergyReactions": [],
  "appointmentParticipants": [],
  "appointments": [],
  "attachments": [],
  "bundleEntries": [],
  "bundles": [
    {
      "createdAt": 2025-08-04T15:45:29.823Z,
      "id": "338c9bba-36fc-436a-84cf-e1fd88e21407",
      "resourceId": undefined,
      "timestamp": undefined,
      "total": undefined,
      "type": "searchset",
    },
  ],
  "carePlanActivities": [],
  "carePlans": [],
  "careTeamParticipants": [],
  "careTeams": [],
  "codeableConcepts": [],
  "codings": [],
  "conditions": [],
  "diagnosticReportResults": [],
  "diagnosticReports": [],
  "encounterParticipants": [],
  "encounters": [],
  "immunizations": [],
  "locations": [],
  "medicationDosages": [],
  "medicationRequests": [],
  "notes": [],
  "observationComponents": [],
  "observations": [],
  "organizationTypes": [],
  "organizations": [],
  "patientContacts": [],
  "patientIdentifiers": [],
  "patientNames": [],
  "patients": [],
  "practitionerQualifications": [],
  "practitioners": [],
  "procedurePerformers": [],
  "procedures": [],
  "references": [],
  "telecoms": [],
}
`;
